resolwe_bio/tools/samplehcluster.py

```diff
@@ -174,24 +141,118 @@ def output_json(result=dict(), fname=None):
 def main():
     """Compute sample hierarchical clustering."""
     args = parse_args()
-    expressions = get_expressions(fnames=args.sample_files, gene_set=args.genes)
-    expressions = transform(expressions, log2=args.log2, normalization=args.normalization)
-    zero_genes = get_zero_genes(expressions)
-    zero_samples = get_zero_samples(expressions)
+
+    if len(args.sample_files) != len(args.sample_ids):
+        msg = 'The number of sample files does not match the number of sample IDs.'
+        print(error(msg))
+        raise ValueError(msg)
+
+    if len(args.sample_files) != len(args.sample_names):
+        msg = 'The number of sample files does not match the number of sample names.'
+        print(error(msg))
+        raise ValueError(msg)
+
+    if len(args.sample_files) < 2:
+        msg = 'Select at least two samples to compute hierarchical clustering of samples.'
+        print(error(msg))
+        raise ValueError(msg)
+
+    if len(args.gene_labels) == 1 and args.distance_metric != 'euclidean':
+        msg = ('Select at least two genes to compute hierarchical clustering of samples with '
```

Comment:
Does it makes sense to add here to error message: "Select more genes or use Euclidean distance metric."?